src/brogue/Monsters.c

```diff
@@ -790,30 +877,24 @@ creature *spawnHorde(short hordeID, short x, short y, unsigned long forbiddenFla
         } while (--failsafe && tryAgain);
     }
 
-    failsafe = 50;
 
     if (x < 0 || y < 0) {
-        i = 0;
-        do {
-            while (!randomMatchingLocation(&(loc[0]), &(loc[1]), FLOOR, NOTHING, (hordeCatalog[hordeID].spawnsIn ? hordeCatalog[hordeID].spawnsIn : -1))
-                   || passableArcCount(loc[0], loc[1]) > 1) {
-                if (!--failsafe) {
-                    return NULL;
-                }
-                hordeID = pickHordeType(depth, 0, forbiddenFlags, 0);
-
-                if (hordeID < 0) {
-                    return NULL;
-                }
+        failsafe = 50;
+        while (failsafe > 0) {
+            if (getRandomHordeSpawnLocation(loc, hordeID, periodic)) {
+                x = loc[0];
+                y = loc[1];
+                break;
             }
-            x = loc[0];
-            y = loc[1];
-            i++;
-
-            // This "while" condition should contain IN_FIELD_OF_VIEW, since that is specifically
-            // calculated from the entry stairs when the level is generated, and will prevent monsters
-            // from spawning within FOV of the entry stairs.
-        } while (i < 25 && (pmap[x][y].flags & (ANY_KIND_OF_VISIBLE | IN_FIELD_OF_VIEW)));
+            hordeID = pickHordeType(depth, 0, forbiddenFlags, 0);
```

Comment:
I just wanted to write it the way it was written before, which apparently works out because the only times spawnHorde is called with x and y < 0 is with requiredFlags = 0 as well. I suppose the 0 should be changed to requiredFlags anyways.

Reply:
I don't mind the 0, to me the weird thing is that spawnHorde will just discard the passed horde type if there is not a valid place to put it in the level, and just pick another one and try to place that. Just seems weird. Could you document this?